packages/python/plotly/plotly/data/__init__.py

```diff
@@ -128,95 +194,216 @@ def election_geojson():
+        return df
+    return df.to_native()
 
 
-def medals_long(indexed=False):
+def medals_long(indexed=False, return_type="pandas"):
     """
     This dataset represents the medal table for Olympic Short Track Speed Skating for the
     top three nations as of 2020.
 
-    Returns:
-        A `pandas.DataFrame` with 9 rows and the following columns:
-        `['nation', 'medal', 'count']`.
-        If `indexed` is True, the 'nation' column is used as the index."""
-    df = _get_dataset("medals").melt(
-        id_vars=["nation"], value_name="count", var_name="medal"
+    Parameters
+    ----------
+    indexed: bool
+        Whether or not the 'nation' column is used as the index.
+        Applicable only if `return_type='pandas'`
+
+    return_type: {'pandas', 'polars', 'pyarrow'}
+        Type of the resulting dataframe
+
+    Returns
+    -------
+    Dataframe of `return_type` type
+        Dataframe with 9 rows and the following columns: `['nation', 'medal', 'count']`.
+        If `indexed` is True, the 'nation' column is used as the index.
+    """
+
+    if indexed and return_type != "pandas":
+        msg = "Cannot set index for backend different from pandas"
+        raise NotImplementedError(msg)
+
+    df = nw.from_native(
+        _get_dataset("medals", return_type=return_type), eager_only=True
+    ).unpivot(
+        index=["nation"],
+        value_name="count",
+        variable_name="medal",
     )
     if indexed:
-        df = df.set_index("nation")
-    return df
+        df = nw.maybe_set_index(df, "nation")
+    return df.to_native()
+
+
+def _get_dataset(d, return_type):
+    """
+    Loads the dataset using the specified backend.
+
+    Notice that the available backends are 'pandas', 'polars', 'pyarrow' and they all
+    have a `read_csv` function. Therefore we can dynamically load the library via
+    `importlib.import_module` and then call `backend.read_csv(filepath)`.
 
+    Parameters
+    ----------
+    d: str
+        Name of the dataset to load.
 
-def _get_dataset(d):
-    import pandas
+    return_type: {'pandas', 'polars', 'pyarrow'}
+        Type of the resulting dataframe
+
+    Returns
+    -------
+    Dataframe of `return_type` type
+    """
     import os
+    from importlib import import_module
 
-    return pandas.read_csv(
-        os.path.join(
-            os.path.dirname(os.path.dirname(__file__)),
-            "package_data",
-            "datasets",
-            d + ".csv.gz",
-        )
+    AVAILABLE_BACKENDS = {"pandas", "polars", "pyarrow"}
+
+    filepath = os.path.join(
+        os.path.dirname(os.path.dirname(__file__)),
+        "package_data",
+        "datasets",
+        d + ".csv.gz",
     )
+    if return_type not in AVAILABLE_BACKENDS:
+        msg = f"Unsupported return_type. Found {return_type}, expected one of {AVAILABLE_BACKENDS}"
+        raise NotImplementedError(msg)
+
+    try:
+        backend = import_module(return_type)
+        return backend.read_csv(filepath)
```

Comment:
It looks like this returns `no attribute 'read_csv'` for `pyarrow` (`pyarrow==18.0.0`):

```
Exception: Unable to read 'carshare' dataset due to: module 'pyarrow' has no attribute 'read_csv'
```

Does it need to be accessed with
```
from pyarrow import csv
```
 https://arrow.apache.org/docs/python/generated/pyarrow.csv.read_csv.html

Reply:
Thanks for the catch, just fixed with the last commit

Reply:
Works well now. Thanks!